api/v1beta2/network_types.go

```diff
@@ -233,16 +233,36 @@ type NetworkSpec struct {
 // IPv6 contains ipv6 specific settings for the network.
 type IPv6 struct {
 	// CidrBlock is the CIDR block provided by Amazon when VPC has enabled IPv6.
+	// Mutually exclusive with IPAMPool.
 	// +optional
 	CidrBlock string `json:"cidrBlock,omitempty"`
 
 	// PoolID is the IP pool which must be defined in case of BYO IP is defined.
+	// Must be specified if CidrBlock is set.
+	// Mutually exclusive with IPAMPool.
 	// +optional
 	PoolID string `json:"poolId,omitempty"`
 
 	// EgressOnlyInternetGatewayID is the id of the egress only internet gateway associated with an IPv6 enabled VPC.
 	// +optional
 	EgressOnlyInternetGatewayID *string `json:"egressOnlyInternetGatewayId,omitempty"`
+
+	// IPAMPool defines the IPAMv6 pool to be used for VPC.
+	// Mutually exclusive with CidrBlock.
+	// +optional
+	IPAMPool *IPAMPool `json:"ipamPool,omitempty"`
+}
+
+// IPAMPool defines the IPAM pool to be used for VPC.
+type IPAMPool struct {
```

Comment:
Not quite sure about this one. In https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/4026 and the subsequent effort to improve networking code, we figured that support for both managed (CAPA creates and deletes it) and unmanaged (CAPA finds it) resources should be very explicit in struct fields. In that issue, the problem was that `ID` could be either filled or empty to denote whether it's an existing or to-be-managed resource.

With the current comments on the fields `ID` and `Name`, as a user, I wouldn't know whether CAPA is going to create the IPAM pool for me. Also, when is which field needed? Can both be given, etc.?

Reply:
That's a good point. We should add //+reqiure or optional and an easier to digest explanation.